docs/walkthrough.md

```diff
@@ -1,17 +1,43 @@
 
->If you haven't installed KEDA and this project, please do so first. Follow instructions [install.md](./install.md) to complete your installation.
+If you haven't installed KEDA and the HTTP Add On (this project), please do so first. Follow instructions [install.md](./install.md) to complete your installation. Before you continue, make sure that you have your `NAMESPACE` environment variable set to the same value as it was when you installed.
 
-## Submitting an `HTTPScaledObject`
+## Creating An Application
 
-You interact with the operator via a CRD called `HTTPScaledObject`. To get an example app up and running, read the notes below and then run the subsequent command from the root of this repository.
+You'll need to install a `Deployment` and `Service` first. You'll tell the add on to begin scaling it up and down after this step. Use this [Helm](https://helm.sh) command to create the resources you need:
 
-- Make sure that your `NAMESPACE` environment variable is set to the same value as what you [install](./install.md)ed with
-  - I recommend using [direnv](https://direnv.net/) to store your environment variables
-- This command will install a simple server that is exposed to the internet using a `LoadBalancer` `Service`. _Do not use this for a production deployment_. Support for `Ingress` is forthcoming in [issue #33](https://github.com/kedacore/http-add-on/issues/33)
+```shell
+helm install xkcd ./charts/xkcd -n ${NAMESPACE}
+```
+
+>To remove the app, run `helm delete xkcd -n ${NAMESPACE}`
+
+## Creating an `HTTPScaledObject`
+
+You interact with the operator via a CRD called `HTTPScaledObject`. This CRD object points the To get an example app up and running, read the notes below and then run the subsequent command from the root of this repository.
 
 ```shell
 kubectl create -f -n $NAMESPACE examples/httpscaledobject.yaml
 ```
+
+>If you'd like to learn more about this object, please see the [`HTTPScaledObject` reference](./ref/http_scaled_object.md).
+
+## Testing Your Installation
+
+You've now installed a web application and activated autoscaling by creating an `HTTPScaledObject` for it. For autoscaling to work properly, HTTP traffic needs to route through the `Service` that the add on has set up. You can use `kubectl port-forward` to quickly test things out:
+
+```shell
+k port-forward svc/xkcd-interceptor-proxy -n ${NAMESPACE} 8080:80 
+```
+
+### Routing to the Right `Service`
+
+As said above, you need to route your HTTP traffic to the `Service` that the add on has created. If you have existing systems - like an ingress controller - you'll need to anticipate the name of these created `Service`s. Each one will be named consistently like so, in the same namespace as the `HTTPScaledObject` and your application (i.e. `$NAMESPACE`):
+
+```shell
+<deployment name>-interceptor-proxy
```

Comment:
Let's also make the interceptor service configurable?

```yaml
kind: HTTPScaledObject
apiVersion: http.keda.sh/v1alpha1
metadata:
    name: xkcd
spec:
    scaleTargetRef:
        deployment: xkcd
        service: xkcd
        port: 8080
    service:
        name: my-interceptor-proxy
        port: 9999
        type: LoadBalancer
```

Or just surface it like this:

```yaml
kind: HTTPScaledObject
apiVersion: http.keda.sh/v1alpha1
metadata:
    name: xkcd
spec:
    scaleTargetRef:
        deployment: xkcd
        service: xkcd
        port: 8080
    service:
        name: my-interceptor-proxy
        spec:
            selector:
              app: MyApp
            ports:
            - protocol: TCP
		      port: 80
		      targetPort: 9376
```

Happy to open an issue if you want.

Reply:
@tomkerkhove yea, can you create an issue for that? One thing I wonder is, should we not have it create a `Service` at all? By not doing so, we won't impose limitations on how they're configured. 

We could then add to the documentation that users can create a `Service` with a specific label selector if they want.

Reply:
What do you think @zroubalik?

I think having it part of the spec might be useful? Users and reading docs doesn't always go that well 😁

Reply:
Very good point :laughing:. Would it be possible to merge this and then either add the `service` field or remove `Service` creation altogether in a follow up (depending on what you and @zroubalik decide)?

Also, if this helps:

In the [scope doc](https://github.com/kedacore/http-add-on/blob/main/docs/scope.md#the-scope-of-the-keda-http-add-on), it says the following is in scope: `Route HTTP traffic from a given source to an arbitrary HTTP server, as far as we need to efficiently accomplish (1)`. That could help the decision. I'd like to prefer having the operator create less things, but I'll defer to what you and @zroubalik think.

Reply:
Moving to another PR is fine for me, but this feels like something we need to have to be able to work imo since this is for the interceptor.

If they have to create the service for KEDA, then it's too much infrastructure imo.

It's all about how you approach it. When I create an HTTPScaledObject you could add a field that surfaces the port + IP (if need be) so that it's all in that CRD when doing `k get HTTPScaledObject` and users don't even know about the service that was created. (not saying we should but a possibility).

Reply:
Ok, yea I tend to agree with that @tomkerkhove . I'm happy to add the fields that surface port, IP, and maybe other fields. This PR is blocking some other work (since it makes such big fundamental changes to the operator). If there's nothing else you'd like to change in it, I'd like us to merge it and I will concurrently open a follow-up PR that adds those fields. What do you think @tomkerkhove @zroubalik ?

Reply:
Fine by me!

Reply:
https://github.com/kedacore/http-add-on/issues/107 is the follow-up issue